### PR TITLE
JS linting

### PR DESCRIPTION
**Pull request by mythmon**
This is going to be a big PR. I recommend looking at individual commit. I broke it into 4 commits.

1. Changes to our eslint config to match more what I think we do/should do and changes to pre-commit so it runs eslint.
2. Fix all the lint errors.
3. Change the eslint config to enforce 2 space indents.
4. Fix all the indentation. 

I'd say at least look over the rules being enforced. If you are feeling picky you could look at how the various lint issues were resolved. A lot of the indentation was mostly automated, so it might be funky.

r?

### CODE REVIEW

**Comment by mythmon** [comment]
PS: I purposely pushed these commits one-at-a-time so Travis would run for all four commits. The odd-numbered ones should fail since they introduce new lint rules, and the even numbered ones should pass, since they fix the lint errors.

File: kitsune/sumo/static/sumo/js/ajaxvote.js [hunk 1]
```diff
@@ -1,183 +1,183 @@
...
+        validate();
+      });
 
-        $radios.bind('change', validate);
+      $radios.bind('change', validate);
 
-        new k.AjaxVote($survey.find('form'), {
-            replaceFormWithMessage: true
-        });
+      new k.AjaxVote($survey.find('form'), { // eslint-disable-line
```
Comment (rehandalal) [inline kitsune/sumo/static/sumo/js/ajaxvote.js]: why are we disabling on this line?

Comment (mythmon) [inline kitsune/sumo/static/sumo/js/ajaxvote.js]: The rule is [Disallow new For Side Effects (no-new)](http://eslint.org/docs/rules/no-new).

File: kitsune/sumo/static/sumo/js/forums.js [hunk 2]
```diff
@@ -1,86 +1,86 @@
...
-            '#reply-content, #id_content',
-            {mediaButton: true});
+  function init() {
+    Marky.createSimpleToolbar(
+      '.editor-tools',
+      '#reply-content, #id_content',
+      {mediaButton: true});
 
-        new k.AjaxPreview($('#preview'));
+    new k.AjaxPreview($('#preview')); // eslint-disable-line
```
Comment (rehandalal) [inline kitsune/sumo/static/sumo/js/forums.js]: oh i'm guessing the linter doesn't like when you create objects and don't assign them?

Comment (rehandalal) [inline kitsune/sumo/static/sumo/js/forums.js]: just saw your reply on the last one

File: kitsune/sumo/static/sumo/js/rickshaw_utils.js [hunk 3]
```diff
@@ -308,8 +308,8 @@
     }
 
     // Rickshaw gets angry when its data isn't sorted.
-    function sortCallback (a, b) {
-      return a.x - b.x;
+    function sortCallback (v1, v2) {
```
Comment (rehandalal) [inline kitsune/sumo/static/sumo/js/rickshaw_utils.js]: does the linter not like `a` and `b` as arguments for functions?

**Comment by rehandalal** [comment]
phew! this looks good. i'm going to click around and see if anything is out of place.

awesome job @mythmon!!

File: kitsune/sumo/static/sumo/js/rickshaw_utils.js (hunk 3 above)
Comment (mythmon) [inline kitsune/sumo/static/sumo/js/rickshaw_utils.js]: Generally, it is fine with them. But [http://eslint.org/docs/rules/no-shadow](http://eslint.org/docs/rules/no-shadow) is on by default, and there is a variable in the parent scope named `b`. There are a few changes like this that are kind of awkward.

**Comment by rehandalal** [comment]
alright i've done some clicking around and seems fine. let's merge this sucker and enter the new age of nicer javascript!

**Pull request merged**